ReactQt/runtime/src/platform.cpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2017-present, Status Research and Development GmbH.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include "platform.h"
+#include "bridge.h"
+#include "eventdispatcher.h"
+
+#include <QQuickItem>
+
+class PlatformPrivate {
+public:
+    PlatformPrivate() {
+#if defined(Q_OS_MACOS)
+        osName = QLatin1String("macos");
+#elif defined(Q_OS_WIN)
+        osName = QLatin1String("windows");
+#elif defined(Q_OS_LINUX)
+        osName = QLatin1String("linux");
+#elif defined(Q_OS_UNIX)
+        osName = QLatin1String("unix");
+#else
+        osName = QLatin1String("unknown");
+#endif
+    }
+
+public:
+    Bridge* bridge = nullptr;
+    QString osName;
+    int major = 0;
```

Comment:
Original `react-native` uses `PlatformConstants` module to store not only os type constants, but also react-native version for runtime checks, so I added it here also.